src/libsyntax/parse/parser.rs

```diff
@@ -5009,7 +5009,7 @@ impl<'a> Parser<'a> {
             |p| {
                 let attrs = p.parse_outer_attributes()?;
                 let lo = p.span.lo;
-                let mut vis = p.parse_visibility()?;
+                let mut vis = p.parse_visibility(true)?;
                 let ty_is_interpolated =
                     p.token.is_interpolated() || p.look_ahead(1, |t| t.is_interpolated());
                 let mut ty = p.parse_ty()?;
```

Comment:
It's not fully on-topic, but could you remove all this cruft remaining from the old visibility syntax, I've missed it somehow.
It should be "`parse_visibility`" -> "`parse_ty`" -> "return the result" now, no extra magic.